test/SafeProtocolMediator.spec.ts

```diff
@@ -0,0 +1,459 @@
+                    SENTINEL_MODULES,
+                ]);
+            });
+        });
+    });
+
+    describe("Execute transaction from module", async () => {
+        async function deployContractsFixture() {
+            const safeProtocolMediator = await (await hre.ethers.getContractFactory("SafeProtocolMediator")).deploy();
+            const safe = await hre.ethers.deployContract("TestExecutor");
+            await safe.setModule(await safeProtocolMediator.getAddress());
+            return { safeProtocolMediator, safe };
+        }
+
+        describe("Module with non-root access", async () => {
+            it("Should not allow non-enabled module to execute tx from a safe", async () => {
+                const { safeProtocolMediator, safe } = await loadFixture(deployContractsFixture);
+                const module = await (await hre.ethers.getContractFactory("TestModule")).deploy();
+                const safeTx = buildSingleTx(user1.address, hre.ethers.parseEther("1"), "0x", BigInt(1), hre.ethers.randomBytes(32));
+                await expect(module.executeFromModule(safeProtocolMediator, safe, safeTx)).to.be.revertedWithCustomError(
+                    safeProtocolMediator,
+                    "MoudleNotEnabled",
+                );
+            });
+
+            it("Should process a SafeTransaction", async function () {
+                const { safeProtocolMediator, safe } = await loadFixture(deployContractsFixture);
+
+                // Enable module
+                const module = await (await hre.ethers.getContractFactory("TestModule")).deploy();
+                const data = safeProtocolMediator.interface.encodeFunctionData("enableModule", [await module.getAddress(), false]);
+                await safe.exec(await safeProtocolMediator.getAddress(), 0, data);
+                const safeAddress = await safe.getAddress();
+                const amount = hre.ethers.parseEther("1");
+                await (
+                    await deployer.sendTransaction({
+                        to: safeAddress,
+                        value: amount,
+                    })
+                ).wait();
+                const safeTx = buildSingleTx(user1.address, hre.ethers.parseEther("1"), "0x", BigInt(1), hre.ethers.randomBytes(32));
+
+                const balanceBefore = await hre.ethers.provider.getBalance(user1.address);
+                const tx = await module.executeFromModule(safeProtocolMediator, safe, safeTx);
+                await tx.wait();
+                const balanceAfter = await hre.ethers.provider.getBalance(user1.address);
+
+                expect(balanceAfter).to.eql(balanceBefore + amount);
+                expect(await hre.ethers.provider.getBalance(safeAddress)).to.eql(0n);
+
+                await expect(tx).to.emit(safeProtocolMediator, "ActionsExecuted").withArgs(safeAddress, safeTx.metaHash, 1);
+            });
+
+            it("Should fail to process a SafeTransaction", async function () {
```

Comment:
The condition is missing from the test description innit? It shouldn't fail just any transaction

Reply:
https://github.com/5afe/safe-protocol/pull/12/commits/b9734c1b9df4483350badb4f2b64a3fef1636b3f